packages/sfui/frameworks/react/hooks/usePagination/usePagination.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+import paginate from 'jw-paginate';
+
+interface UsePaginationParameters {
+  totalItems: number;
+  currentPage?: number;
+  pageSize?: number;
+  maxPages?: number;
+}
+
+export function usePagination({ totalItems, currentPage = 1, pageSize = 10, maxPages = 1 }: UsePaginationParameters) {
+  const [pagination, setPagination] = useState(paginate(totalItems, currentPage, pageSize, maxPages));
+  const [selectedPage, setSelectedPage] = useState(currentPage);
+
+  useEffect(() => {
+    setSelectedPage(currentPage);
+  }, [currentPage]);
+
+  useEffect(() => {
+    setPagination(paginate(totalItems, selectedPage, pageSize, maxPages));
+  }, [totalItems, selectedPage, pageSize, maxPages]);
+
+  const onPageChange = (page: number) => {
+    setSelectedPage(page);
+  };
+
+  const next = () => {
+    setSelectedPage((pagination.currentPage += 1));
+  };
+
+  const prev = () => {
+    setSelectedPage((pagination.currentPage -= 1));
+  };
+
+  return {
+    totalPages: pagination.totalPages,
+    maxVisiblePages: maxPages,
+    itemsPerPage: pageSize,
+    pages: pagination.pages,
+    selectedPage,
+    endPage: pagination.endPage,
+    startPage: pagination.startPage,
+    next,
+    prev,
+    onPageChange,
```

Comment:
can we do the same here as well ?:) https://github.com/vuestorefront/storefront-ui/pull/2535#discussion_r1154407516

Reply:
sure :)